src/auth/csrf.guard.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Injectable,
+  CanActivate,
+  UnauthorizedException,
+  ExecutionContext,
+  Logger,
+} from '@nestjs/common';
+import { FastifyRequest } from 'fastify';
+import { FormMode, LoginRequest } from './api/login.request';
+
+@Injectable()
+export class CsrfGuard implements CanActivate {
+  private static readonly CSRF_COOKIE_HEADER = '_csrf';
+  private readonly logger = new Logger(CsrfGuard.name);
+
+  async canActivate(context: ExecutionContext) {
+    this.logger.debug('Called canActivate');
+
+    const request: FastifyRequest = context.switchToHttp().getRequest();
+
+    try {
+      const body: LoginRequest = request.body as LoginRequest;
+      if (body?.op === FormMode.CSRF) {
+        const { csrf: csrfToken } = body;
+        const csrfCookie = request.cookies[CsrfGuard.CSRF_COOKIE_HEADER];
+        if (csrfCookie) {
+          const token = decodeURIComponent(csrfCookie);
+          if (token === csrfToken) {
+            return true;
+          }
+        }
+        this.throwError();
```

Comment:
Please refactor this code to reduce its complexity (there are too many branches)